jetstream/cli.py

```diff
@@ -56,28 +54,71 @@ class AllType:
 
 class ExecutorStrategy(Protocol):
     project_id: str
-    dataset_id: str
 
-    def __init__(self, project_id: str, dataset_id: str, *args, **kwargs) -> None:
+    def __init__(self, project_id: str, *args, **kwargs) -> None:
         ...
 
-    def execute(self, worklist: Iterable[Tuple[str, AnalysisSpec, datetime]]) -> bool:
+    def execute(
+        self,
+        worklist: Iterable[Tuple[str, datetime]],
+        configuration_map: Mapping[str, TextIO] = {},
```

Comment:
nit: i know we never [mutate the default argument](https://towardsdatascience.com/python-pitfall-mutable-default-arguments-9385e8265422) but writing this as `Optional[Mapping[str, TextIO]] = None` would make that more clear